### ISSUE TITLE
CDT 11.0.0

### ISSUE THREAD

**Issue by jonahgraham** [eclipse-cdt/cdt#77]
This is the Release plan and TODO list for the release in the title. CDT makes its simrel contribution on +1 day in the release cycle (normally Monday by 10pm UTC).

## Steps for M1

Items before M1 +1 day:
- [ ] Create release on [PMI](https://projects.eclipse.org/projects/tools.cdt)
- [ ] Create [New and Noteworthy page](https://wiki.eclipse.org/CDT/User/NewInCDT) for the release
- [ ] Prepare for the new release with all these updates: (Note: As I do this work I check the boxes, but I only check the top level one on this line once it is merged)
  - [ ] Update cdt.target to point to new dependency versions
  - [ ] Synchronize CDT.setup to cdt.target
  - [ ] Update API baseline to last release and resolve any API errors
  - [ ] Synchronize CDT.setup for API baseline
  - [ ] Update `help-docs-eclipserun-repo` in root pom.xml and bump versions of all the docs plug-ins
  - [ ] Increment version of all feature.xml, pom.xml and any other place full version is used. (Easiest way is global find and replace, e.g. `s/9.9.0/9.10.0/g` and review changes.) (See [this gerrit](https://git.eclipse.org/r/#/c/154068/) for a past example)
  - [ ] Update `comparator.repo` in root pom.xml to last release and if there have been any changes since the branch, bump versions of MANIFEST.MFs (If your local java version is the same as the docker container, or if you build in the docker container, a command line like `mvn verify -DskipTests -P baseline-compare-and-replace -P build-standalone-debugger-rcp` is good to test this has worked and nothing needs updating.)
    - [ ] The following need their bundle service segment updated all the time for various reasons:
      - [ ] `org.eclipse.cdt.debug.application` because the version number is encoded in the bundle
      - [ ] `org.eclipse.cdt.debug.dap` because it pulls and builds with yarn
  - [ ] Update Maven versions (check [CI job](https://ci.eclipse.org/cdt/view/Releng/job/cdt-check-maven-plugin-updates/))
  - [ ] Update `simrel-site` in root pom.xml
  - [ ] Remove old API error filters if they are no longer relevant. In the Problems view you can filter to see only these types of problems by selecting only "API Problems/Unused API Problem Filter Problem". (See [this gerrit](https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/179197) for a past example)
- [ ] Ensure the build is stable - it is always better to release a "Green Dot" build
- [ ] Ensure all previous [Endgame issues](https://github.com/eclipse-cdt/cdt-infra/labels/endgame) are done.

Items on M1 +0 day:
- [ ] If tracking most recent platform, look for announcement and 
  - [ ] update cdt.target to most recent platform
  - [ ] update dependencies in MANIFEST.MF to require that platform (see https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/149208 as an example)
- [ ] If tracking most recent [cdt-gdb-adapter](https://github.com/eclipse-cdt/cdt-gdb-adapter), update the [yarn.lock](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/yarn.lock)/[package.json](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/package.json) with `yarn upgrade cdt-gdb-adapter`

Items on M1 +1 day:
- [ ] Run [CQ license check](https://ci.eclipse.org/cdt/view/all/job/cdt-master-run-dash-licenses/) and make sure no new CQs need to be filed - this does not need to be done for every milestone, but needs to be done if there are dependency changes.
- [ ] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform M1, then STANDALONE can be checked, otherwise do it on +4
- [ ] Smoke test the release by installing it into platform's SDK build.
- [ ] Add description to the promote-a-build job and the job it promoted.
- [ ] Mark promoted job as Keep forever
- [ ] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [ ] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on M1 +4 day (or shortly after):
- [ ] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 

## Steps for M2

Items on M2 +0 day:
- [ ] If tracking most recent platform, look for announcement and 
  - [ ] update cdt.target to most recent platform
  - [ ] update dependencies in MANIFEST.MF to require that platform (see https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/149208 as an example)
- [ ] If tracking most recent [cdt-gdb-adapter](https://github.com/eclipse-cdt/cdt-gdb-adapter), update the [yarn.lock](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/yarn.lock)/[package.json](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/package.json) with `yarn upgrade cdt-gdb-adapter`

Items on M2 +1 day:
- [ ] Run [CQ license check](https://ci.eclipse.org/cdt/view/all/job/cdt-master-run-dash-licenses/) and make sure no new CQs need to be filed - this does not need to be done for every milestone, but needs to be done if there are dependency changes.
- [ ] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform M2, then STANDALONE can be checked, otherwise do it on +4
- [ ] Smoke test the release by installing it into platform's SDK build.
- [ ] Add description to the promote-a-build job and the job it promoted.
- [ ] Mark promoted job as Keep forever
- [ ] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [ ] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on M2 +4 day (or shortly after):
- [ ] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 

## Steps for M3

Items before M3 +1 day:
- [ ] Schedule a release or progress review if it is needed (generally >= 12 months since the last one). Consult [EDP](https://www.eclipse.org/projects/dev_process/) for guidance.

Items on M3 +0 day:
- [ ] If tracking most recent platform, look for announcement and 
  - [ ] update cdt.target to most recent platform
  - [ ] update dependencies in MANIFEST.MF to require that platform (see https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/149208 as an example)
- [ ] If tracking most recent [cdt-gdb-adapter](https://github.com/eclipse-cdt/cdt-gdb-adapter), update the [yarn.lock](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/yarn.lock)/[package.json](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/package.json) with `yarn upgrade cdt-gdb-adapter`

Items on M3 +1 day:
- [ ] Run [CQ license check](https://ci.eclipse.org/cdt/view/all/job/cdt-master-run-dash-licenses/) and make sure no new CQs need to be filed - this does not need to be done for every milestone, but needs to be done if there are dependency changes.
- [ ] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform M3, then STANDALONE can be checked, otherwise do it on +4
- [ ] Smoke test the release by installing it into platform's SDK build.
- [ ] Add description to the promote-a-build job and the job it promoted.
- [ ] Mark promoted job as Keep forever
- [ ] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [ ] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on M3 +4 day (or shortly after):
- [ ] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 

## Steps for RC1

Items before RC1 +1 day:
- [ ] Ensure release entry on [PMI](https://projects.eclipse.org/projects/tools.cdt) "Release Date" section it says the appropriate "This release is part of Eclipse IDE ??????". 
- [ ] Ensure there are no API errors

Items on RC1 +0 day:
- [ ] If tracking most recent platform, look for announcement and 
  - [ ] update cdt.target to most recent platform
  - [ ] update dependencies in MANIFEST.MF to require that platform (see https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/149208 as an example)
- [ ] If tracking most recent [cdt-gdb-adapter](https://github.com/eclipse-cdt/cdt-gdb-adapter), update the [yarn.lock](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/yarn.lock)/[package.json](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/package.json) with `yarn upgrade cdt-gdb-adapter`

Items on RC1 +1 day:
- [ ] Run [CQ license check](https://ci.eclipse.org/cdt/view/all/job/cdt-master-run-dash-licenses/) and make sure no new CQs need to be filed - this does not need to be done for every milestone, but needs to be done if there are dependency changes. **Note it is probably too late to get a CQ resolved in time for release at this point, so consider mitigation too.**
- [ ] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform RC1, then STANDALONE can be checked, otherwise do it on +4
- [ ] Smoke test the release by installing it into platform's SDK build.
- [ ] Add description to the promote-a-build job and the job it promoted.
- [ ] Mark promoted job as Keep forever
- [ ] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [ ] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on RC1 +4 day (or shortly after):
- [ ] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 

## Steps for RC2

These steps should be done before RC2 release, they can be completed at anytime between the last branch creation/release and RC2.

Items before RC2 +1 day:
- [ ] Run [CQ license check](https://ci.eclipse.org/cdt/view/all/job/cdt-master-run-dash-licenses/) and make sure no new CQs need to be filed - this does not need to be done for every milestone, but needs to be done if there are dependency changes. **Note it is probably too late to get a CQ resolved in time for release at this point, so consider mitigation too.**
- [ ] Create branch for CDT release
- [ ] Create new [Jenkinsfiles](https://github.com/eclipse-cdt/cdt-infra/tree/master/jenkins/pipelines/cdt) for new release and create jobs
- [ ] Make sure [New and Noteworthy page](https://wiki.eclipse.org/CDT/User/NewInCDT) is up to date

Items on RC2 +0 day:
- [ ] If tracking most recent platform, look for announcement and 
  - [ ] update cdt.target to most recent platform
  - [ ] update dependencies in MANIFEST.MF to require that platform (see https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/149208 as an example)
- [ ] If tracking most recent [cdt-gdb-adapter](https://github.com/eclipse-cdt/cdt-gdb-adapter), update the [yarn.lock](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/yarn.lock)/[package.json](https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/tree/debug/org.eclipse.cdt.debug.dap/debug-servers/package.json) with `yarn upgrade cdt-gdb-adapter`

Items on RC2 +1 day:
- [ ] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform RC2, then STANDALONE can be checked, otherwise do it on +4
- [ ] Smoke test the release by installing it into platform's SDK build.
- [ ] Add description to the promote-a-build job and the job it promoted.
- [ ] Mark promoted job as Keep forever
- [ ] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [ ] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on RC2 +4 day (or shortly after):
- [ ] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 

## Steps for Release

Items for quiet week (between RC2 and release):
- [ ] Make sure documentation is part of simrel's [help](http://help.eclipse.org).
- [ ] Finalize the [testing spreadsheet for the release](https://docs.google.com/spreadsheets/d/1xj6lEmLWpZC7rwHv9f7t-2FWLGViiSdYQK7SSDEdf6s/edit#gid=0)

Items to be done 2 days before release:
- [ ] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to releases (run twice so standalone is promoted too if needed)
- [ ] Add description to the promote-a-build job and the job it promoted.
- [ ] Unmark as keep all old Milestone and RC jobs
- [ ] Update or create [composites](https://github.com/eclipse-cdt/cdt-infra/tree/master/download/releases) in cdt-infra in preparation for going public on release day
  - [ ] Include the update to latest URL https://download.eclipse.org/tools/cdt/releases/latest/ to point to latest release

Release day:
- [ ] Verify we get a GO on the cross-project-issues-dev mailing list
- [ ] [Promote the files to download](https://ci.eclipse.org/cdt/view/Releng/job/promote-files-to-download/) - this makes the release public
- [ ] Tag the release. Example: `git tag -a CDT_9_4_2 2cdc63eae52c8952c0d2543e1e31ca6e25225c4a -m"CDT 9.4.2" && git push origin CDT_9_4_2`
- [ ] Update https://www.eclipse.org/cdt/ (current status) and https://www.eclipse.org/cdt/downloads.php
- [ ] Update Marketplace entries:
  - [ ] https://marketplace.eclipse.org/node/3799227/edit (Complete C/C++ IDE)
  - [ ] https://marketplace.eclipse.org/node/1373403/edit (CDT)
  - [ ] https://marketplace.eclipse.org/node/1687099/edit (Terminal)
- [ ] Make an announcement on cdt-dev. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33618.html)

Items after the release:
- [ ] Delete download.eclipse.org/tools/cdt/builds for this release
    - [ ] Visit https://archive.eclipse.org/tools/cdt/ and press login if the checkboxes and Archive button are not visible
    - [ ] Visit https://archive.eclipse.org/tools/cdt/ and delete "builds" if present
    - [ ] Visit https://download.eclipse.org/tools/cdt/ and archive "builds"
    - [ ] Optional visit https://archive.eclipse.org/tools/cdt/ and delete "builds"
- [ ] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon) to point to the release as it is not probably pointing to the cdt/builds directory. (e.g. https://git.eclipse.org/r/#/c/154853/)
- [ ] Review and discard any "kept forever" builds, especially milestone and superseded RC builds for cdt-master, and jobs on cdt-verify-test-cdt-other-pipeline that were kept around to be tested.
- [ ] Update `help-docs-eclipserun-repo` in root `pom.xml` (remote `-I-builds` from path) on new branch

**Comment by jonahgraham** [comment]
As discussed at our [last call](https://hackmd.io/@jonahgraham/cdt-call-november-8-2022), I planned to prepare a draft release with plenty of time before the CDT 11.0.0 release.

The release is next Wednesday (Dec 7th).
 
Committers should be able to see the CDT 11.0.0 draft release here: https://github.com/eclipse-cdt/cdt/releases (please edit as you see fit, but do not publish it). PS I can't confirm that anyone else can see it, perhaps only I can see the draft?

In addition I added a link to a GitHub search for noteworthy issues and PRs to the N&N document:
 https://github.com/eclipse-cdt/cdt/blob/main/NewAndNoteworthy/CDT-11.0.md#noteworthy-issues-and-pull-requests

Please comment here on the issue with your thoughts.

**Comment by jld01** [comment]
Hi @jonahgraham, do you think it's preferable/valuable to list the full set of PRs for milestone 11.0.0 on the release anouncement page as opposed to just the noteworthy PRs? Many of the non-noteworthy PRs will be of little interest to typical readers of the announcement. A counterargument is that every PR represents someone's effort towards the release and should be recognized.

**Comment by jonahgraham** [comment]
I had a similar thought - I like the autogenerated part(0), especially the New Contributors section(1). But I found it to be of little value(2), which is why I hid it behind the collapsible section(3).

(0) I imagine that as GitHub evolves the autogeneration will get more interesting too.

(1) In this release the "new contributors" section is a little off because we just moved to GitHub. You, me and @Torbjorn-Svensson aren't listed because we are listed in the [10.7.1 release page](https://github.com/eclipse-cdt/cdt/releases/tag/CDT_10_7_1)

(2) I agree with you about the value of recognizing someone's contribution. It can be a nice ego boost to see your name/contribution listed up front like that. The other value is it shows life of the project a little, similar purpose to the charts on [PMI](https://projects.eclipse.org/projects/tools.cdt/who)


PS I don't want to spend time manually maintaining a list. I think it is the auto-generated list, a link to the [search](https://github.com/eclipse-cdt/cdt/issues?q=is%3Aclosed+label%3Anoteworthy+milestone%3A11.0.0), or I copy and paste the N&N into the release page.

**Comment by jonahgraham** [comment]
> Update or create [composites](https://github.com/eclipse-cdt/cdt/tree/main/releng/download/releases) in preparation for going public on release day

I made a mistake in the above step, which is fixed in fb081b99759c77686ec51eca04bd89d577c36c83 but there is caching on download.eclipse.org which the p2 repo will present as slightly broken until the caches catch up.

They seem to be caught up for me already, but the GitHubActions seems to be seeing the old versions

**Comment by jonahgraham** [comment]
CDT 11.0.0 is done, onto CDT 11.1.0 in #202 

PS. If we need a CDT 11.0.1 we can create that at short notice.
